File: meme_inator_front/lib/features/feeds/domain/entities/value_objects/duration_window_vo.dart
import 'package:equatable/equatable.dart';
import 'package:meme_inator_front/features/post/domain/entities/post_entity.dart';

/// Duration window value object (one bucket of posts)
class DurationWindowVo extends Equatable {
  final String label;
  final DateTime startDate;
  final DateTime endDate;
  final List<PostEntity> posts;

  const DurationWindowVo({
    required this.label,
    required this.startDate,
    required this.endDate,
    required this.posts,
  });

  @override
  List<Object?> get props => [label, startDate, endDate, posts];
}

File: meme_inator_back/apps/feeds/domain/entities/duration_window_vo.py
from dataclasses import dataclass, field
from datetime import datetime
from typing import List, Optional

from apps.posts.domain.entities.post_entity import PostEntity


@dataclass
class DurationWindow:
    """
    Represents one duration window (group).
    - window_start: inclusive (datetime)
    - window_end: exclusive (datetime)
    - posts: list of PostEntity (hydrated or raw depending on place in pipeline)
    - window_key: string identifier (e.g., '2025-02-06' or '2025-W06')
    """
    window_start: datetime
    window_end: datetime
    posts: List[PostEntity]
    window_key: Optional[str] = None
